### ISSUE TITLE
Export Canvas as PNG does decrease the quality of the image

### ISSUE THREAD

**Issue by styriandev** [pixijs/pixijs#8757]
It seems like the renderer.extract.canvas function is decreasing the quality of the canvas. Is there a possability to change that behaviour by another function or some parameter?

The prepared example that demonstrates that issue is located under [Example (Stackblitz)](https://stackblitz.com/edit/angular-xmu6se-pixijs-dil7ja?file=src/app/app.component.ts). If you check that out you will find a canvas with some example object + a button that exports the canvas as a PNG via renderer.extract.canvas. You will notice that the quality of the exported PNG has some problems if you zoom to the circle:
![image](https://user-images.githubusercontent.com/46619772/196905182-0da56ef4-a606-420d-a2e0-64708bf1d49f.png)

The interesting thing here is that if I export the canvas via right click "Save image as" the quality is much better (there are less pixalated borders):
![image](https://user-images.githubusercontent.com/46619772/196905391-586de69f-a0ae-40af-b90f-f465bb8c39fe.png)

The problem is that this solution is not acceptable as in my application there is also such a button that will trigger the export. Also it seems like the problem is getting bigger with bigger canvas sizes. If I export the canvas as a PNG inside the application I am getting the following pixelated image:
![image](https://user-images.githubusercontent.com/46619772/196906949-573dfbff-ee95-4221-83e1-74560afb6d7e.png)

In the actual canvas that looks like that:
![image](https://user-images.githubusercontent.com/46619772/196907064-7ef50722-5e24-4e0d-ae12-1b80122bba0e.png)

### Expected Behavior
Actually I would expect the exported PNG to look like the canvas, so it should not decrease the quality of the images. 

### Current Behavior
There are pixelated parts which are not pixelated on the canvas. 

### Steps to Reproduce
Open stackblitz, press on the button "Save PNG" to export the pixelated image. Then right click on the canvas, select "Save image as" and compare the 2 images.

[Example (Stackblitz)](https://stackblitz.com/edit/angular-xmu6se-pixijs-dil7ja?file=src/app/app.component.ts) 
### Environment

- **`pixi.js` version**: 6.5.6
- **OS & Version**: Windows 10
- **Running Example**: [Example (Stackblitz)](https://stackblitz.com/edit/angular-xmu6se-pixijs-dil7ja?file=src/app/app.component.ts)

**Comment by styriandev** [comment]
Thanks for pointing that workaround. Unfortunally this does not work as it is only exporting one specific part of the canvas (I guess the part that is visible). As we have a Zoom feature as well as a scroll to specific parts of the canvas this could lead to different screenshots for the same canvas which is not the solution that we need.
However, if I scroll out and create a screenshot of the whole canvas the quality of the image is the following:
 
![image](https://user-images.githubusercontent.com/46619772/197464126-754d6df8-9ef1-493f-9c9f-57ba30f98e16.png)

So as you see not really good. I have used the following command to get the blob (after the app.renderer was called):
this.app.renderer.plugins.extract.canvas().toBlob(blob =>{...}, 'image/png', 0);

I have also tried to change the last parameter to 1 but this does not help.

**Comment by orange4glace** [comment]
I tried this and seems work?

```ts
  downloadSpriteAsImage(renderer, sprite, fileName) {
      var a = document.createElement('a');
      document.body.append(a);
      a.download = fileName;
      document.getElementById('a').toBlob(blob => {
        const url = URL.createObjectURL(blob);
        a.href = url;
        a.click();
        a.remove();
      }, 'image/png');
  }
```

FYI https://webglfundamentals.org/webgl/lessons/webgl-tips.html

**Comment by styriandev** [comment]
Accessing the canvas and getting the blob from the canvas just returns a blank image

**Comment by domis86** [comment]
Maybe solution like this? : 

1. Zoom to desired level.
2. "Move" canvas to see top left corner of whole scene.
3. Extract canvas image and save it in some "buffer".
4. "Move" canvas to the right.
5. Repeat points 3 and 4 until you reach right side of scene.
6. Move canvas to see next chunk of scene on the left.
7. Repeat points 3.4,6,7 until you scan whole scene.
8. Somehow join extracted chunks into one image 😃 

In other words: extract chunks of scene and glue those images together int one image.

**Comment by SuperSodaSea** [comment]
This issue should have been fixed in [pixi.js@6.5.9](https://github.com/pixijs/pixijs/releases/tag/v6.5.9) (See #9102 and #9103), so I'm going to close this issue. If the problem still exists, you are welcome to reopen it.

**Comment by styriandev** [comment]
> This issue should have been fixed in [pixi.js@6.5.9](https://github.com/pixijs/pixijs/releases/tag/v6.5.9) (See #9102 and #9103), so I'm going to close this issue. If the problem still exists, you are welcome to reopen it.

Thank you for solving this issue and also merging that PR into version 6.5.9!
